packages/client/src/__tests__/integration/happy/not-so-exhaustive-schema-mongo/__snapshots__/binary.test.ts.snap

```diff
@@ -2277,10 +2277,15 @@ export namespace Prisma {
     emit: 'stdout' | 'event'
   }
 
-  export type GetLogType<T extends LogLevel | LogDefinition> = T extends LogDefinition ? T['emit'] extends 'event' ? T['level'] : never : never
-  export type GetEvents<T extends any> = T extends Array<LogLevel | LogDefinition> ?
-    GetLogType<T[0]> | GetLogType<T[1]> | GetLogType<T[2]> | GetLogType<T[3]>
-    : never
+  export type CheckIsLogLevel<T> = T extends LogLevel ? T : never;
+
+  export type GetLogType<T> = T extends LogDefinition
+    ? (T['emit'] extends 'event' ? CheckIsLogLevel<T['level']> : never)
+    : CheckIsLogLevel<T>;
```

Comment:
So, about this. String literals are a shorthand for `{ level: "...", emit: "stdout" }`, so it was intentional that `never` was returned in this case, since [according to the docs](https://www.prisma.io/docs/orm/prisma-client/observability-and-logging/logging) it's either events or logs in stdout but not both.

However, this is not the actual behavior. You will indeed receive both events and console logs when using either the shorthand or the full `emit: "stdout"` form.

Technically this is a bug, as it contradicts both the types and the documentation. However, since we're doing a major release anyway, I'm fine with changing the semantics the way you propose and updating the types and the docs to match the actual behavior because it makes more sense to me too. In that case, though, we shouldn't restrict the `LogDefinition` case to `T['emit'] extends 'event'` either.

Reply:
Do you mean changing it from `T['emit'] extends 'event' ? CheckIsLogLevel<T['level']> : never` to just `CheckIsLogLevel<T['level']>`, right?

Also did I understand correctly that you want me to change the docs too?

Reply:
> Do you mean changing it from T['emit'] extends 'event' ? CheckIsLogLevel<T['level']> : never to just CheckIsLogLevel<T['level']>, right?

Yes

> Also did I understand correctly that you want me to change the docs too? I'm happy to do it

That would be amazing, but up to you, we can also do this. Probably a small note that tells that `emit: 'stdout'` logs are also emitted as events is sufficient.